lib/core/src/lib/common/utils/date-fns-adapter.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DateFnsAdapter } from '@angular/material-date-fns-adapter';
+import { DateFnsUtils } from './date-fns-utils';
+import { Inject, Injectable, Optional } from '@angular/core';
+import { MAT_DATE_FORMATS, MAT_DATE_LOCALE, MatDateFormats } from '@angular/material/core';
+import { UserPreferenceValues, UserPreferencesService } from '../services/user-preferences.service';
+
+/**
+ * Date-fns adapter with moment-to-date-fns conversion.
+ *
+ * Automatically switches locales based on user preferences.
+ * Supports custom display format.
+ *
+ * @example
+ *
+ * Add the following to the component `providers` section
+ *
+ * providers: [
+ *  { provide: MAT_DATE_FORMATS, useValue: ADF_FORM_DATE_FORMATS },
+ *  { provide: DateAdapter, useClass: AdfDateFnsAdapter }
+ * ]
+ *
+ * Setting custom format
+ *
+ * constructor(private dateAdapter: DateAdapter<Date>) {}
+ *
+ * ngOnInit() {
+ *   const adapter = this.dateAdapter as AdfDateFnsAdapter;
+     adapter.displayFormat = '<custom date-fns format>';
+ * }
+ */
+@Injectable()
+export class AdfDateFnsAdapter extends DateFnsAdapter {
+    private _displayFormat?: string = null;
+
+    get displayFormat(): string | null {
+        return this._displayFormat;
+    }
+
+    set displayFormat(value: string | null) {
+        this._displayFormat = value ? DateFnsUtils.convertMomentToDateFnsFormat(value) : null;
+    }
+
+    constructor(
+        @Optional() @Inject(MAT_DATE_LOCALE) matDateLocale: any,
```

Comment:
Can we avoid using `any` here?

Reply:
I'll see what is the options, took it from here https://github.com/angular/components/blob/main/src/material-date-fns-adapter/adapter/date-fns-adapter.ts#L55 but latest TS flags empty object as a warning, and in moment adapter Angular is using "any"... but maybe I can find a type

Reply:
used `Locale` type from date-fns as that's what expected